### PR TITLE
DSDEGP-1434 Roll clio logs daily, with max size.

### PR DESCRIPTION
**Pull request by danxmoran**
### Description

Our logs already roll, but based on size. Henry recommended time-based instead, to make finding things easier if we need to search through the log files on disk.

----

### Checklist (never delete this)

Never delete this, it is our record that procedure was followed. If you find that for whatever reason one of the checklist points doesn't apply to your PR, you can leave it unchecked but please add an explanation below.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: Ensure that you have added or updated tests
- [x] **Submitter**: If you're adding/switching libraries or otherwise changing the design, update the [design documentation](https://broadinstitute.atlassian.net/wiki/pages/viewpage.action?pageId=114531509).
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * Add notes on what you've tested
* Review cycle:
  * Team may comment on PR at will
  * **Reviewer assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to reviewer** for further feedback
- [ ] **Submitter**: Verify all tests go green, including CI tests
- [ ] **Submitter**: Squash commits and merge to develop
- [ ] **Submitter**: Delete branch after merge
- [ ] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [ ] **Submitter**: Move the JIRA issue to QA once this checklist is completed

### CODE REVIEW

File: clio-server/config/clio-logback.xml.ctmpl [hunk 1]
```diff
@@ -4,15 +4,14 @@
...
         <file>{{$log_dir}}/clio.log</file>
 
-        <rollingPolicy class="ch.qos.logback.core.rolling.FixedWindowRollingPolicy">
-            <fileNamePattern>{{$log_dir}}/clio.%i.log.gz</fileNamePattern>
-            <minIndex>1</minIndex>
-            <maxIndex>5</maxIndex>
-        </rollingPolicy>
+        <rollingPolicy class="ch.qos.logback.core.rolling.TimeBasedRollingPolicy">
+            <!-- roll over daily, inferred by logback from the filename pattern -->
+            <fileNamePattern>{{$log_dir}}/clio-%d{yyyy-MM-dd}.log</fileNamePattern>
```
Comment (pshapiro4broad) [inline clio-server/config/clio-logback.xml.ctmpl]: We should probably compress the logs as well, when rolled. I think changing the pattern to `.log.gz` will enable this automatically.

File: clio-server/config/clio-logback.xml.ctmpl [hunk 2]
```diff
@@ -4,15 +4,14 @@
...
 
-        <triggeringPolicy class="ch.qos.logback.core.rolling.SizeBasedTriggeringPolicy">
-            <maxFileSize>10MB</maxFileSize>
-        </triggeringPolicy>
+            <!-- keep 30 days' worth of logs, up to 5GB -->
+            <maxHistory>30</maxHistory>
+            <totalSizeCap>5GB</totalSizeCap>
```
Comment (pshapiro4broad) [inline clio-server/config/clio-logback.xml.ctmpl]: I worry about any log size cap at all until we have an idea about how much space 30 days of logs should take up. On the other hand, if logs are compressed we will probably be fine.

**Review by pshapiro4broad — APPROVED** [review]
👍  with a few minor comments

**Comment by danxmoran** [comment]
@ktibbett do these look like good settings to you?

**Comment by danxmoran** [comment]
Taking KT's verbal +1 as an official +1 for pullapprove.

**Pull request merged**